pkg/cloudprovider/suite_test.go

```diff
@@ -76,7 +77,7 @@ func TestCloudProvider(t *testing.T) {
 }
 
 var _ = BeforeSuite(func() {
-	env = coretest.NewEnvironment(coretest.WithCRDs(apis.CRDs...), coretest.WithCRDs(v1alpha1.CRDs...))
+	env = coretest.NewEnvironment(coretest.WithCRDs(apis.CRDs...), coretest.WithCRDs(v1alpha1.CRDs...), coretest.WithFieldIndexers(coretest.NodeProviderIDFieldIndexer(ctx)))
```

Comment:
Was looking to ensure we don't have to setup an indexer on this in prod. Appears core already sets one up, which is what I expected:
- https://github.com/kubernetes-sigs/karpenter/blob/8c7f0cb5f626270213dc7fdfccf7c1a8c22f538f/pkg/operator/operator.go#L239

Also, didn't spot anything special for this indexer in a quick pass. They do have some other indexers added though:
https://github.com/aws/karpenter-provider-aws/blob/24d63454f069ab6b35538f502aaf43399fec73c3/pkg/operator/operator.go#L305